### PR TITLE
Use Material 3

### PR DESCRIPTION
**Pull request by dev-Fabi**
Replace Material2 with Material3 and refactor improve UI of some screens.

### CODE REVIEW

File: androidApp/src/main/java/org/datepollsystems/waiterrobot/android/MainActivity.kt [hunk 1]
```diff
@@ -38,12 +38,12 @@ class MainActivity : AppCompatActivity() {
         // Switch from SplashScreenTheme to AppTheme
         setTheme(R.style.Theme_WaiterRobot_Main)
 
-        checkForUpdate()
-
         setContent {
             RootView(vm, ::onAppThemeChange)
         }
 
+        checkForUpdate()
```
Comment (Dafnik) [inline androidApp/src/main/java/org/datepollsystems/waiterrobot/android/MainActivity.kt:45]: If there is a reason for this to be after the `setContent` we should add a comment perhaps to explain this a little bit further

**Review by Dafnik — APPROVED** [review]
The whole UI refactoring looks generally speaking good, but I'm definitely not able to provide valuable insights by just reading compose code.

The only thing I have to add is that the Pull to refresh behaviour seems very complicated to do :D

**Comment by Dafnik** [comment]
One more thing, is there a reason there is a left & right padding besides the chips in this screenshot? [screenshot](https://private-user-images.githubusercontent.com/73596295/309551933-3ac58306-8e29-4006-896b-6d57ac38ebc5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk1NDg1MjUsIm5iZiI6MTcwOTU0ODIyNSwicGF0aCI6Ii83MzU5NjI5NS8zMDk1NTE5MzMtM2FjNTgzMDYtOGUyOS00MDA2LTg5NmItNmQ1N2FjMzhlYmM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA0VDEwMzAyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNjZTAxYjBiMWJmNWEyZTg0MmVmYTI4MDM3MzE1MDU4MjgxZDI3MzYxNjg4ZDc0NGFlMWZlNmY3OTg5NmNkMTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ELKavW8YwTApGAZonMqJBJ1TbnwR-kbDHvng14b0Un4). Instead of it being aligned with the rest?

File: androidApp/src/main/java/org/datepollsystems/waiterrobot/android/MainActivity.kt (hunk 1 above)
Comment (dev-Fabi) [reply androidApp/src/main/java/org/datepollsystems/waiterrobot/android/MainActivity.kt:45]: No reason - I think that was just moved when I tried something and then did not revert it :)

**Pull request merged**